panther_analysis_tool/main.py

```diff
@@ -398,6 +399,90 @@ def upload_analysis(args: argparse.Namespace) -> Tuple[int, str]:
     return 0, ""
 
 
+def confirm_analysis_exists(args: argparse.Namespace, analysis_id_list: list) -> list:
+    validation_payload = {"listDetections": {"ids": analysis_id_list, "fields": ["id"]}}
+    client = get_client(args.aws_profile, "lambda")
+
+    validation = client.invoke(
+        FunctionName="panther-analysis-api",
+        InvocationType="RequestResponse",
+        LogType="None",
+        Payload=json.dumps(validation_payload),
+    )
+
+    validation_string = json.loads(validation["Payload"].read().decode("utf-8"))["body"]
+    analysis_found_count = json.loads(validation_string)["paging"]["totalItems"]
+
+    if len(analysis_id_list) != analysis_found_count:
+        analysis_confirmed = []
+        analysis_found = {}
+        json_output = json.loads(validation_string)
+
+        for detection in json_output["detections"]:
+            analysis_found[detection["id"]] = detection
+        # Build a list of analysis we've confirmed are in Panther
+        for analysis in analysis_id_list:
+            if analysis not in analysis_found:
+                logging.info("%s was not found, skipping...", analysis)
+            else:
+                analysis_confirmed.append(analysis)
+        return analysis_confirmed
+    return analysis_id_list
+
+
+# pylint: disable=too-many-locals
+def delete_analysis(args: argparse.Namespace) -> Tuple[int, str]:
+
+    client = get_client(args.aws_profile, "lambda")
+    analysis_id_list = args.analysis_id
+    payload: dict = {"deleteDetections": {"entries": []}}
+
+    # Validate the Detection that we are deleting exists in Panther
+    # If nothing was found bail out, otherwise prepare to delete what has been confirmed
+    analysis_confirmed = confirm_analysis_exists(args, analysis_id_list)
+
+    if len(analysis_confirmed) == 0:
+        logging.error("No matching analysis found, exiting")
+        return 1, ""
+
+    analysis_id_list = analysis_confirmed
```

Comment:
wondering if you could just set the return to analysis_id_list? 

```suggestion
    payload: dict = {"deleteDetections": {"entries": []}}

    # Validate the Detection that we are deleting exists in Panther
    # If nothing was found bail out, otherwise prepare to delete what has been confirmed
    analysis_id_list = confirm_analysis_exists(args, args.analysis_id)

    if len(analysis_id_list) == 0:
        logging.error("No matching analysis found, exiting")
        return 1, ""
```

Reply:
Done and done